django_paypal/models.py

```diff
@@ -1,187 +1,158 @@
-    payment_id = models.CharField(_("payment id"), max_length=255, unique=True)  # id created by paypal
-    intent = models.CharField(_("intent"), max_length=255)  # sale
-    experience_profile_id = models.CharField(_("intent"), max_length=255, blank=True)  # sale
-    note_to_payer = models.CharField(_("intent"), max_length=165, blank=True)  # sale
-    state = models.CharField(_("state"), max_length=255, blank=True)
-    payment_method = models.CharField(_("payment method"), max_length=255, blank=True)
-    custom = models.CharField(_("payment method"), max_length=255, blank=True)
-    payer_id = models.CharField(_("payer id"), max_length=255, blank=True)
-    transaction_fee = models.DecimalField(_("transaction fee"), max_digits=9, decimal_places=2, default=Decimal(0))
-    related_resource_id = models.CharField(_("related resource id"), max_length=255, blank=True, null=True)
-    related_resource_state = models.CharField(_("related resource state"), max_length=255, blank=True, null=True)
-    initial_response_object = models.TextField(_("initial post response"), null=True, blank=True)
-    update_response_object = models.TextField(_("updated get response"), null=True, blank=True)
 
-    failure_reason = models.CharField(_("failure reason"), max_length=255, blank=True)
+class PaypalOrder(models.Model):
+    order_id = models.CharField(_('Order ID'), max_length=255, unique=True)
+    status = models.CharField(_('Status'), max_length=255, blank=True)
 
-    link = models.URLField(_("link"), blank=True)
-    approve_link = models.URLField(_("approve link"), blank=True)
-    execute_link = models.URLField(_("execute link"), blank=True)
 
-    created_at = models.DateTimeField(_("created at"), auto_now_add=True)
-    last_modified = models.DateTimeField(_("last modified"), auto_now=True)
+class PaypalAPIResponse(models.Model):
+    order = models.ForeignKey(PaypalOrder, related_name='api_responses', on_delete=models.CASCADE)
+    url = models.CharField(_('URL'), max_length=255, blank=True)
+    response_data = JSONField(_('Response Data'))
+    created_at = models.DateTimeField(auto_now_add=True)
+
+
+class PaypalAPIPostData(models.Model):
+    order = models.ForeignKey(PaypalOrder, related_name='api_posts', on_delete=models.CASCADE)
+    url = models.CharField(_('URL'), max_length=255, blank=True)
+    post_data = JSONField(_('Post Data'))
+    created_at = models.DateTimeField(auto_now_add=True)
+
+
+class PaypalWebhook(models.Model):
+    webhook_id = models.CharField(_('Webhook ID'), max_length=255, unique=True)
+    url = models.CharField('URL', max_length=255, unique=True)
+    events = JSONField(_('Active Webhook Events'))
+
+
+# The following models solely exist to keep payments made in version <0.3.0 stored in the database.
+# They are not used in the current version of the app and have been created using /v1/payments.
+# They might get removed in a future version.
+class LegacyPaypalPayment(models.Model):
```

Comment:
Maybe add a deprecated decorator.
```
from warnings import deprecated

@deprecated("Deprecated model for deprecated PayPal API")
class LegacyPaypalPayment(models.Model): ...
```

Reply:
I would, if there would be a decorator for that in python standard libraries